zeroconf/src/linux/event_loop.rs

```diff
@@ -16,10 +17,9 @@ pub struct AvahiEventLoop<'a> {
 impl<'a> TEventLoop for AvahiEventLoop<'a> {
     /// Polls for new events.
     ///
-    /// Internally calls `ManagedAvahiSimplePoll::iterate(0)`, the `timeout` parameter does not
-    /// currently do anything in the Avahi implementation.
-    fn poll(&self, _timeout: Duration) -> Result<()> {
-        self.poll.iterate(0);
-        Ok(())
+    /// The `timeout` parameter defines the maximum time to sleep, but it will return earlier if
+    /// an event occurs.
+    fn poll(&self, timeout: Duration) -> Result<()> {
+        self.poll.iterate(timeout.as_millis().try_into().unwrap_or(-1))
```

Comment:
Just as a note, this means that if the timeout is > `i32::MAX` millis, then it will become an infinite wait.

Considering that's about 23 days, I'm not sure anyone is going to run into this in practice, but it could be documented as a 'feature'.

Reply:
Could you please add a note to the docs that the default behavior (-1) will block indefinitely until new events are registered.